url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/jgraph/example/fastgraph/FastGraphUI.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2005, David Benson
+ *
+ * See LICENSE file in distribution for licensing details of this source file
+ */
+package com.jgraph.example.fastgraph;
+
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Insets;
+import java.awt.Point;
+import java.awt.RenderingHints;
+import java.awt.Stroke;
+import java.awt.geom.Point2D;
+import java.awt.geom.Rectangle2D;
+
+import org.jgraph.graph.CellView;
+import org.jgraph.graph.EdgeView;
+import org.jgraph.graph.GraphConstants;
+import org.jgraph.plaf.basic.BasicGraphUI;
+
+public class FastGraphUI extends BasicGraphUI {
+	/**
+	 * Paints the renderer of <code>view</code> to <code>g</code> at
+	 * <code>bounds</code>. Recursive implementation that paints the children
+	 * first.
+	 * <p>
+	 * The reciever should NOT modify <code>clipBounds</code>, or
+	 * <code>insets</code>. The <code>preview</code> flag is passed to the
+	 * renderer, and is not used here.
+	 */
+	public void paintCell(Graphics g, CellView view, Rectangle2D bounds,
+			boolean preview) {
+		if (view != null && view instanceof FastEdgeView) {
+			// First Paint View
+			if (bounds != null) {
+				boolean selected = graph.isCellSelected(view.getCell());
+				Graphics2D g2 = (Graphics2D) g;
+				Stroke stroke = g2.getStroke();
+				Object rendererHint = g2
+						.getRenderingHint(RenderingHints.KEY_STROKE_CONTROL);
+
+				g2.setRenderingHint(RenderingHints.KEY_STROKE_CONTROL,
+						RenderingHints.VALUE_STROKE_PURE);
+				int c = BasicStroke.CAP_BUTT;
+				int j = BasicStroke.JOIN_MITER;
+				float lineWidth = 1.0f;
+				g2.setStroke(new BasicStroke(lineWidth, c, j));
+				// TODO
+				g2.setColor(Color.BLACK);//(getForeground());
+				int n = ((EdgeView)view).getPointCount();
+				Point2D lastPoint = null;
+				for (int i = 0; i < n; i++) {
+					Point2D currentPoint = ((EdgeView) view).getPoint(i);
+					if (i != 0) {
+						g2.drawLine((int) lastPoint.getX(), (int) lastPoint.getY(),
+								(int) currentPoint.getX(),
+								(int) currentPoint.getY());
+					}
+					lastPoint = currentPoint;
+				}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 56:64, 68:76.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.